Modules/Forum/classes/class.ilForumNotification.php

```diff
@@ -277,11 +277,21 @@ public static function getCachedNodeData($ref_id)
     {
         if (!array_key_exists($ref_id, self::$node_data_cache)) {
             global $DIC;
-            self::$node_data_cache[$ref_id] = $DIC->repositoryTree()->getSubTree(
+            $node_data = $DIC->repositoryTree()->getSubTree(
                 $DIC->repositoryTree()->getNodeData($ref_id),
                 true,
                 'frm'
             );
+            $node_data = array_filter($node_data, function ($forum_node) use ($DIC, $ref_id) {
+                // filter out forum if a grp lies in the path (#0027702)
+                foreach ($DIC->repositoryTree()->getNodePath($forum_node['child'], $ref_id) as $path_node) {
+                    if ($path_node['child'] !== $ref_id && $path_node['type'] == 'grp') {
+                        return false;
```

Comment:
it doesn't work. This causes an error if you delete a crs/grp-member. 

Whoops\Exception\ErrorException thrown with message "Invalid argument supplied for foreach()"

Stacktrace:
#15 Whoops\Exception\ErrorException in /var/www/databay_ilias/Modules/Forum/classes/class.ilForumNotification.php:257
#14 ilErrorHandling:handlePreWhoops in /var/www/databay_ilias/Modules/Forum/classes/class.ilForumNotification.php:257
#13 ilForumNotification:checkForumsExistsDelete in /var/www/databay_ilias/Modules/Forum/classes/class.ilForumAppEventListener.php:392
#12 ilForumAppEventListener:handleEvent in /var/www/databay_ilias/Services/EventHandling/classes/class.ilAppEventHandler.php:187
#11 ilAppEventHandler:raise in /var/www/databay_ilias/Services/Membership/classes/class.ilParticipants.php:832
#10 ilParticipants:delete in /var/www/databay_ilias/Services/Membership/classes/class.ilParticipants.php:1046
#9 ilParticipants:deleteParticipants in /var/www/databay_ilias/Services/Membership/classes/class.ilMembershipGUI.php:719
#8 ilMembershipGUI:deleteParticipants in /var/www/databay_ilias/Services/Membership/classes/class.ilMembershipGUI.php:392
#7 ilMembershipGUI:executeCommand in /var/www/databay_ilias/Services/UICore/classes/class.ilCtrl.php:210
#6 ilCtrl:forwardCommand in /var/www/databay_ilias/Modules/Course/classes/class.ilObjCourseGUI.php:2258
#5 ilObjCourseGUI:executeCommand in /var/www/databay_ilias/Services/UICore/classes/class.ilCtrl.php:210
#4 ilCtrl:forwardCommand in /var/www/databay_ilias/Services/Repository/classes/class.ilRepositoryGUI.php:348
#3 ilRepositoryGUI:show in /var/www/databay_ilias/Services/Repository/classes/class.ilRepositoryGUI.php:310
#2 ilRepositoryGUI:executeCommand in /var/www/databay_ilias/Services/UICore/classes/class.ilCtrl.php:210
#1 ilCtrl:forwardCommand in /var/www/databay_ilias/Services/UICore/classes/class.ilCtrl.php:175
#0 ilCtrl:callBaseClass in /var/www/databay_ilias/ilias.php:20


-- GET Data --

ref_id                   : 72
cmdClass                 : ilcoursemembershipgui
cmd                      : post
cmdNode                  : u1:jo:82
baseClass                : ilrepositorygui
fallbackCmd              : confirmDeleteParticipants
rtoken                   : bcf4da0e50b8c1ceb9a39ee75cdd1d0b
lang                     : de
limit                    : 50

Reply:
Hmm that's weird. I can't think of a reason, why this should not be an array. Also, line 257 is not a foreach loop in my code. Could it be, that you're on a different version?

Reply:
@theodortruffer I'll discuss this with @nmatuschek  This is post probably NOT an issue.